packages/application-shell/src/test-utils/test-utils.tsx

```diff
@@ -272,19 +336,25 @@ const renderApp = (
     actionRights, // <-- deprecated option, use `{ project: { allAppliedActionRights } }`
     dataFences, // <-- deprecated option, use `{ project: { allAppliedDataFences } }`
     dataLocale = 'en',
-    ApolloProviderComponent = MockedApolloProvider,
+    ApolloProviderComponent = ApolloMockProvider,
     // gtm-context
     gtmTracking = defaultGtmTracking,
     // forwarding to @testing-library/react
     ...renderOptions
-  } = {}
+  }: Partial<TRenderAppOptions> = {}
 ) => {
-  const mergedUser = mergeOptional(defaultUser, user);
-  const mergedProject = mergeOptional(defaultProject, project);
-  const mergedEnvironment = mergeOptional(defaultEnvironment, environment);
-  const mergedGtmTracking = mergeOptional(defaultGtmTracking, gtmTracking);
+  const mergedUser = mergeOptional<
+    ProviderProps<AdditionalEnvironmentProperties>['user']
+  >(defaultUser, user);
+  const mergedProject = mergeOptional<
+    ProviderProps<AdditionalEnvironmentProperties>['project']
+  >(defaultProject, project);
+  const mergedEnvironment = mergeOptional<
+    ProviderProps<AdditionalEnvironmentProperties>['environment']
+  >(defaultEnvironment, environment);
+  const mergedGtmTracking = { ...defaultGtmTracking, ...gtmTracking };
 
-  const ApplicationProviders = ({ children }) => (
+  const ApplicationProviders = (props: TApplicationProvidersProps) => (
     <IntlProvider locale={locale}>
       <ApolloProviderComponent mocks={mocks} addTypename={addTypename}>
         <ConfigureFlopFlip
```

Comment:
FYI: the types are currently failing here because of this: https://github.com/tdeekens/flopflip/pull/890